packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -993,7 +987,7 @@ mixin RenderAnimatedOpacityMixin<T extends RenderObject> on RenderObjectWithChil
     _alpha = ui.Color.getAlphaFromOpacity(opacity.value);
     if (oldAlpha != _alpha) {
       final bool? didNeedCompositing = _currentlyNeedsCompositing;
-      _currentlyNeedsCompositing = _alpha! > 0 && _alpha! < 255;
+      _currentlyNeedsCompositing = _alpha! > 0;
```

Comment:
Here we use inequality for alpha test, other places use equality testing. Perhaps all tests on alpha should use tests that are safe in the presence of out-of-range values like this?

Reply:
Also, there are range asserts sprinkled around, but it is hard to tell from a PR diff if they are consistently protecting every line of code.

Reply:
We use the following code from dart:ui to calculate alpha

```
  static int getAlphaFromOpacity(double opacity) {
    assert(opacity != null);
    return (opacity.clamp(0.0, 1.0) * 255).round();
  }
```

It's guarnateed to not go out of bounds.